airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -237,10 +237,7 @@ private void scanFile(MediaFile file, MusicFolder musicFolder, Date lastScanned,
         indexManager.index(file);
 
         if (file.isDirectory()) {
-            for (MediaFile child : mediaFileService.getChildrenOf(file, true, false, false, false)) {
-                scanFile(child, musicFolder, lastScanned, albumCount, genres, isPodcast);
-            }
-            for (MediaFile child : mediaFileService.getChildrenOf(file, false, true, false, false)) {
+            for (MediaFile child : mediaFileService.getChildrenOf(file, true, true, false, false)) {
```

Comment:
Are we sure that this doesn't have side effects?  It seems unlikely that this was done explicitly separately unless there was a reason to.